### PR TITLE
nim js: --browser, --livereload, --jshtml

### PR DESCRIPTION
**Pull request by timotheecour**
* `nim r -b:js --browser main` now does what you'd expect: opens a browser with an html page containing a js script obtained from the jsgen'd output of `nim js main`; this naturally extends `r` (these already worked before this PS: `nim doc -r main`, `nim r -b:js main`)
* livereload support
* make `nim --fullhelp` a bit more organized by grouping the js specific options together

these simplify testing code intended for browser use (as opposed to nodejs use)

## example
```
nim doc -b:js --doccmd:'--browser -d:nimExperimentalJsfetch -d:nimExperimentalAsyncjsThen -r' lib/std/jsfetch.nim
```
this opens a browser, runs the runnableExamples and displays the fetch output (which is intended for browser only, and doesn't work with nodejs). The fact that we have those extra flags needed here is unrelated to this PR and can be simplified later

### CODE REVIEW

File: lib/std/private/browserutils.nim [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+from std/strutils import `%`
+import std/compilesettings
+
+# proc serveJsBrowser*(fileJs: string, port: int) =
+proc serveJsBrowser*(fileJs: string) =
+  let port = 7031 # PRTEMP
+  let jsname = "input.js"
+  let html = fmt"""
+<!DOCTYPE html>
+<html>
```
Comment (juancarlospaco) [inline lib/std/private/browserutils.nim:20]:
```suggestion
<html>
<head>
<title>Nim</title>
</head>
```

Must have non empty title to be HTML valid.

File: lib/std/private/browserutils.nim [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+proc serveJsBrowserLivereload*(fileJs: string, fileHtml = "", port = portOff) =
+  ##[
+  [livereload-js - npm](https://www.npmjs.com/package/livereload-js)
+  ]##
+
+  let content = fileJs.readFile
+  var extra = livereloadString(port)
+  let html = fmt"""
+<!DOCTYPE html>
+<html>
```
Comment (juancarlospaco) [inline lib/std/private/browserutils.nim]:
```suggestion
<html>
<head>
<title>Nim</title>
</head>
```

File: lib/std/private/browserutils.nim [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  let fileHtml = dir / "index.html"
+  let fileJs2 = dir / jsname
+  writeFile(fileHtml, html)
+  removeFile(fileJs2)
+  createSymlink(fileJs, fileJs2)
+  echo (fileHtml,)
+  let url = fmt"http://localhost:{port}/"
+  # python3 -m http.server 7031 --directory /tmp/d06/
+  openDefaultBrowser(fileHtml)
+  let status = execShellCmd(fmt"python3 -m http.server {port} --directory {dir.quoteShell}")
```
Comment (juancarlospaco) [inline lib/std/private/browserutils.nim:46]: This adds Python dependency for Windows and Mac (usually has a very old Python),
should be documented.

Comment (ringabout) [reply lib/std/private/browserutils.nim:46]: this dependency is not reasonable, you should serve them with asynchttpserver

Comment (juancarlospaco) [reply lib/std/private/browserutils.nim:46]: We need an `python3 -m http.server` equivalent in stdlib.

File: lib/std/private/browserutils.nim (hunk 1 above)
Comment (timotheecour) [reply lib/std/private/browserutils.nim:20]: done

File: lib/std/private/browserutils.nim (hunk 2 above)
Comment (timotheecour) [reply lib/std/private/browserutils.nim]: done

**Comment by Araq** [comment]
Closing this one. Not because it's bad, but because these "Draft" PRs are piling up, please concentrate your efforts on fewer different aspects and open Pull requests when you think they are review-ready. Of course, starting with an RFC first is also appreciated.

**Pull request closed without merging**